app/controllers/signatures_controller.rb

```diff
@@ -141,7 +141,7 @@ def retrieve_signature
     @signature = Signature.find(signature_id)
     @petition = @signature.petition
 
-    unless @petition.visible?
+    unless @petition.visible? || controller_name == "petitioners"
```

Comment:
Actually, inheriting from `SponsorsController` may be a better option since it deals with hidden petitions, etc. already.